synapse/storage/databases/main/event_push_actions.py

```diff
@@ -287,23 +286,23 @@ def _get_unread_counts_by_pos_txn(
                     notify_count=notif_count, unread_count=unread_count
                 )
 
-            # XXX All threads should have the same stream ordering?
-            max_summary_stream_ordering = max(
-                summary_stream_ordering, max_summary_stream_ordering
-            )
```

Comment:
Something seems a little off in this code -- should each thread have it's own summary stream ordering or each room? The code needs to be audited either way to ensure that we're doing the proper thing here.